### PR TITLE
Update ci to use girder 3

### PR DESCRIPTION
**Pull request by kotfic**

### CODE REVIEW

**Comment by kotfic** [comment]
Note that I manually triggered the integration tests on this branch to verify that these ran correctly ([see workflow](https://circleci.com/workflow-run/05b63a63-ca6d-4b0d-9cfb-b006d9c38486)).  I've since removed the commit as these should now be running nightly.

File: tests/integration/Dockerfile.girder_worker [hunk 1]
```diff
@@ -17,7 +17,7 @@ VOLUME /girder_worker
 # Make sure remote debugging is available
 RUN pip install rpdb
 # Make sure we have the newest girder_client (on 2.x-maintenance)
-RUN pip install -U "git+https://github.com/girder/girder.git@2.x-maintenance#egg=girder_client&subdirectory=clients/python"
+RUN pip install -U "git+https://github.com/girder/girder.git@master#egg=girder_client&subdirectory=clients/python"
```
Comment (zachmullen) [inline tests/integration/Dockerfile.girder_worker]: I'd vote to change this to `pip install --pre -U girder-client`

File: tests/integration/Dockerfile.girder_worker-py3 [hunk 2]
```diff
@@ -17,7 +17,7 @@ VOLUME /girder_worker
 # Make sure remote debugging is available
 RUN pip3 install rpdb
 # Make sure we have the newest girder_client (on 2.x-maintenance)
-RUN pip install -U "git+https://github.com/girder/girder.git@2.x-maintenance#egg=girder_client&subdirectory=clients/python"
+RUN pip install -U "git+https://github.com/girder/girder.git@master#egg=girder_client&subdirectory=clients/python"
```
Comment (zachmullen) [inline tests/integration/Dockerfile.girder_worker-py3]: Same here

File: tests/integration/scripts/girder_entrypoint.sh [hunk 3]
```diff
@@ -1,4 +1,31 @@
...
+
+# If integration_test_endpoints/setup.py exists then we've
+# mounted girder_worker at run time,  make sure it
+# is properly installed before continuing
+if [ -e /girder_worker/tests/integration/common_tasks/setup.py ]; then
+    $PIP uninstall -y integration_test_endpoints
+    $PIP install -e /girder_worker/tests/integration/integration_test_endpoints/
+fi
+
+# TODO: convert this to `girder serve` once complete
```
Comment (zachmullen) [inline tests/integration/scripts/girder_entrypoint.sh]: This TODO seems to have been fulfilled

**Pull request merged**